app/main.py

```diff
@@ -1 +1,18 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import NotWearingMaskError
+from app.errors import VaccineError
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
Returning immediately when a `VaccineError` is caught prevents checking other friends. Consider accumulating a count of unvaccinated friends or handling this case differently to ensure all friends are checked.